### PR TITLE
[full-ci][tests-only] Add retry option

### PR DESCRIPTION
**Pull request by SwikritiT**
## Description
This PR adds retry option in: 
-  make command in `makeNodeGenerate()`
- make command in `makeGoGenerate()`
- make command in `build()`

## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/3962

## How Has This Been Tested?

- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by phil-davis** [comment]
Note: https://github.com/owncloud-ci/retry/issues/7

Maybe `retry` should be added into the `owncloudci/golang` image. Let's see what @xoxys thinks.

File: .drone.star [hunk 1]
```diff
@@ -1598,7 +1598,8 @@ def makeGoGenerate(module):
             "name": "generate go",
             "image": OC_CI_GOLANG,
             "commands": [
-                "%s ci-go-generate" % (make),
+                "curl -SsfL -o /usr/local/bin/retry https://raw.githubusercontent.com/owncloud-ci/retry/master/retry && chmod +x /usr/local/bin/retry",
```
Comment (phil-davis) [inline .drone.star:1601]: golang images have "retry" in them now - https://github.com/owncloud-ci/golang/pull/91
So this line should not be needed.

File: .drone.star [hunk 2]
```diff
@@ -1844,7 +1845,8 @@ def build():
             "name": "build",
             "image": OC_CI_GOLANG,
             "commands": [
-                "make -C ocis build",
+                "curl -SsfL -o /usr/local/bin/retry https://raw.githubusercontent.com/owncloud-ci/retry/master/retry && chmod +x /usr/local/bin/retry",
```
Comment (phil-davis) [inline .drone.star:1848]: golang images have "retry" in them now - https://github.com/owncloud-ci/golang/pull/91
So this line should not be needed.

File: .drone.star (hunk 1 above)
Comment (SwikritiT) [reply .drone.star]: Removed!

**Review by phil-davis — APPROVED** [review]

File: .drone.star [hunk 3]
```diff
@@ -1599,7 +1599,7 @@ def makeNodeGenerate(module):
                 "CHROMEDRIVER_SKIP_DOWNLOAD": "true",  # install fails on arm and chromedriver is a test only dependency
             },
             "commands": [
-                "%s ci-node-generate" % (make),
+                "retry -t 3 '%s ci-node-generate'" % (make),
```
Comment (phil-davis) [inline .drone.star:1602]:
```suggestion
                "/usr/local/bin/retry -t 3 '%s ci-node-generate'" % (make),
```

Maybe the PATH list in these images does not look in `/usr/local/bin` - maybe we have to give the full path?

**Review by phil-davis — APPROVED** [review]

**Pull request merged**